src/aks-preview/azext_aks_preview/_helpers.py

```diff
@@ -374,3 +374,21 @@ def check_is_monitoring_addon_enabled(addons, instance):
     except Exception as ex:  # pylint: disable=broad-except
         logger.debug("failed to check monitoring addon enabled: %s", ex)
     return is_monitoring_addon_enabled
+
+
+def filter_hard_taints(node_initialization_taints: List[str]) -> List[str]:
+    filtered_taints = []
+    for taint in node_initialization_taints:
+        if not taint:
+            continue
+        # Parse the taint to get the effect
+        taint_parts = taint.split(":")
+        if len(taint_parts) == 2:
+            effect = taint_parts[-1].strip()
+            # Only keep taints with soft effects (PreferNoSchedule)
+            if effect.lower() == "prefernoschedule":
+                filtered_taints.append(taint)
```

Comment:
From a user experience perspective, I think adding a warning-level log here can better inform users that the input they provided has been modified.

Reply:
Added a warning with `logger.warning(...)` in this function - would that work?